src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -96,10 +96,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
 
             If node.RelaxationLambdaOpt IsNot Nothing Then
-                returnValue = node.Update(VisitExpressionNode(node.RelaxationLambdaOpt),
+
+                If _inExpressionLambda AndAlso
+                    NoParameterRelaxation(DirectCast(node.Operand, BoundLambda).LambdaSymbol, node.RelaxationLambdaOpt.LambdaSymbol) Then
```

Comment:
The new change should handle this scenario by falling back to preexisting behavior. Regardless what native compiler does in this case, I think it goes beyond this fix.

Will add a test to ensure this scenario though.